### PR TITLE
Release 2.0.0 beta.15

### PR DESCRIPTION
**Pull request by sampaccoud**
### Added

- Add main content placeholder to programs list view
- Create a function checking username in social auth pipeline
- Extend social-auth middleware to use a page to display error
- Add `urls` property to `UserSerializer`. The urls are built with a new setting `MAIN_LMS_USER_URLS` which contains custom links to access to LMS profile views from richie.
- Add a useMatchMedia hook to easily show/hide react component through media query
- Add LimitBrowserCacheTTLHeaders middleware to limit the browser cache TTL

### Changed

- Allow default CKEditor settings on "maincontent" placeholders
- Remove copyright notice from footer in base template and allow customizing it
- Allow customizing the side pages created by the demo site
- Make demo site work without licences placeholder
- Allow empty content on glimpse plugins
- Switch from deprecated tslint to eslint js linter and update codebase accordingly.
- Improve Course Search UX by triggering scroll up after courses are retrieved

### Fixed

- Frontend did not report error to Sentry
- Fix how EDX_USER_PROFILE_TO_DJANGO default value is set
- Use variables for button colors in language selector so it fits in with themes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -240,11 +240,11 @@ test-back: ## run back-end tests
 
 # -- Internationalization
 crowdin-download: ## download translated message from Crowdin
-	@$(COMPOSE_RUN_CROWDIN) download translations
+	@$(COMPOSE_RUN_CROWDIN) download -c crowdin/config.yml
 .PHONY: crowdin-download
 
 crowdin-upload: ## upload source translations to Crowdin
-	@$(COMPOSE_RUN_CROWDIN) upload sources
+	@$(COMPOSE_RUN_CROWDIN) upload -c crowdin/config.yml
```
Comment (lunika) [inline Makefile:247]:
```suggestion
	@$(COMPOSE_RUN_CROWDIN) upload sources -c crowdin/config.yml
```

File: tests_e2e/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "richie-tests-e2e",
-  "version": "2.0.0-beta.14",
+  "version": "2.0.0-",
```
Comment (lunika) [inline tests_e2e/package.json:3]: why removing beta tag ?

File: website/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "richie-education-docs",
-  "version": "2.0.0-beta.14",
```
Comment (lunika) [inline website/package.json:3]: Ditto

**Review by lunika — COMMENTED** [review]
You forget to update the circleci config to use crowdin image version 3 and also don't forget to add new environment variable in circleci richie project.

File: Makefile (hunk 1 above)
Comment (sampaccoud) [reply Makefile:247]: both seem to work but ok.

File: website/package.json (hunk 3 above)
Comment (sampaccoud) [reply website/package.json:3]: search/replace fail :pray:

File: tests_e2e/package.json (hunk 2 above)
Comment (sampaccoud) [reply tests_e2e/package.json]: search/replace fail :pray:

**Comment by sampaccoud** [comment]
@lunika thanks for your review. I corrected the glitches.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
+## [2.0.0-beta.15] - 2020-09-06
```
Comment (jmaupetit) [inline CHANGELOG.md:11]: Time flies...

```suggestion
## [2.0.0-beta.15] - 2020-10-06
```

**Review by jmaupetit — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (sampaccoud) [reply CHANGELOG.md]: Sorry Daddy. Fixed.

**Review by lunika — APPROVED** [review]

**Pull request merged**